File: dtos/image_dto.go
package dtos

type ImageDto struct {
	URLs   ImageURLs   `json:"urls"`
	Author ImageAuthor `json:"author"`
}

type ImageURLs struct {
	Small   string `json:"small"`
	Regular string `json:"regular"`
	Raw     string `json:"raw"`
}

type ImageAuthor struct {
	Name       string `json:"name"`
	Username   string `json:"username"`
	ProfileURL string `json:"profileURL"`
}

File: web/src/data/dtos/ingredients/ImageDto.ts
export class ImageDto {
    urls: ImageURLs = new ImageURLs();
    author: ImageAuthor = new ImageAuthor();

    constructor(obj: Partial<ImageDto>) {
        Object.assign(this, obj);
    }
}

class ImageURLs {
    small: string = '';
    regular: string = '';
    raw: string = '';
}

class ImageAuthor {
    name: string = '';
    username: string = '';
    profileURL: string = '';
}
